src/main/java/reproducer/BreakingUpdateReproducer.java

```diff
@@ -57,6 +59,28 @@ public BreakingUpdateReproducer(ResultManager resultManager) {
         }
     }
 
+    /**
+     * Iterate through a list of breaking updates and attempt to reproduce if not already attempted.
+     * @param breakingUpdates the list of breaking updates to reproduce.
+     */
+    public void reproduceAll(File[] breakingUpdates) {
+        if (breakingUpdates != null) {
+            for (File breakingUpdate : breakingUpdates) {
+                try {
+                    BreakingUpdate bu = JsonUtils.readFromFile(breakingUpdate.toPath(), BreakingUpdate.class);
+                    if (bu.getReproductionStatus().equals("not_attempted")) {
+                        reproduce(bu);
+                    }
+                } catch (RuntimeException e) {
+                    log.error("An exception occurred while reproducing the breaking update in " +
+                            breakingUpdate.getName(), e);
+                }
+            }
+        } else {
+            throw new RuntimeException("The provided directory containing breaking updates is empty.");
```

Comment:
Maybe this check is better handled in `Main`, before we pass the File array to the `reproduceAll` method?

Reply:
Thanks for the comment, will update it.